src/test/java/io/github/bobocodebreskul/context/registry/SimpleBeanDefinitionRegistryTest.java

```diff
@@ -0,0 +1,105 @@
+            testBeanName);
+    String actualMessage = exception.getMessage();
+
+    assertTrue(actualMessage.contains(expectedMessage));
+  }
+
+  @Test
+  void givenBeanNameThenRemoveBeanDefinition() {
+    BeanDefinition beanDefinitionMock1 = Mockito.mock(BeanDefinition.class);
+    BeanDefinition beanDefinitionMock2 = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock1);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName2", beanDefinitionMock2);
+
+    assertNotNull(beanDefinitionRegistry.getBeanDefinition("testBeanName1"));
+    beanDefinitionRegistry.removeBeanDefinition("testBeanName");
+    beanDefinitionRegistry.removeBeanDefinition("testBeanName1");
+    assertNull(beanDefinitionRegistry.getBeanDefinition("testBeanName1"));
+    assertNotNull(beanDefinitionRegistry.getBeanDefinition("testBeanName2"));
+  }
+
+  @Test
+  void givenBeanNameThenGetBeanDefinition() {
+    BeanDefinition beanDefinitionMock1 = Mockito.mock(BeanDefinition.class);
+    BeanDefinition beanDefinitionMock2 = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock1);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName2", beanDefinitionMock2);
+
+    assertNull(beanDefinitionRegistry.getBeanDefinition("testBeanName"));
+    assertSame(beanDefinitionMock1, beanDefinitionRegistry.getBeanDefinition("testBeanName1"));
+    assertSame(beanDefinitionMock2, beanDefinitionRegistry.getBeanDefinition("testBeanName2"));
+  }
+
+  @Test
+  void givenBeanNameThenContainsBeanDefinition() {
+    BeanDefinition beanDefinitionMock = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName2", beanDefinitionMock);
+
+    assertFalse(beanDefinitionRegistry.containsBeanDefinition("testBeanName"));
+    assertTrue(beanDefinitionRegistry.containsBeanDefinition("testBeanName1"));
+    assertTrue(beanDefinitionRegistry.containsBeanDefinition("testBeanName2"));
+  }
+
+  @Test
+  void whenRegisterBeanDefinitionsThenGetRegisteredBeanDefinitionNames() {
+    assertEquals(0, beanDefinitionRegistry.getBeanDefinitionNames().length);
+
+    BeanDefinition beanDefinitionMock = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName2", beanDefinitionMock);
+    String[] beanDefinitionNames = beanDefinitionRegistry.getBeanDefinitionNames();
+
+    assertEquals(2, beanDefinitionNames.length);
+    assertThat(beanDefinitionNames).contains("testBeanName1", "testBeanName2");
+  }
+
+  @Test
+  void whenRegisterBeanDefinitionsThenGetRegisteredBeanDefinitionCount() {
```

Comment:
I would advice writing one more test to check the count after bean definition removal.